### ISSUE TITLE
Chat - 'task for' task copy words not translated on copy

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29082]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Keep language as spanish
3. Open report which has task message in it
4. Hover or long press on task message and copy to clipboard
5. Paste and observe that it is in english
6. Change language and again copy the message
7. Paste and observe that 'task for ' part of the message is not translated with language change

## Expected Result:
App should translate client side set text on change in language

## Actual Result:
App does not translate 'task for' text set client side for task message on copy to clipboard of task message

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.78.3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/9d23f70f-81c0-48b0-9c19-565c67b858d6

https://github.com/Expensify/App/assets/93399543/706ecc4f-3edd-45ca-8f70-78c522f9e60b

https://github.com/Expensify/App/assets/93399543/9b52c04b-a155-440b-b9cb-cbee6ff94517

https://github.com/Expensify/App/assets/93399543/6831dc2c-471e-441e-acd1-761808f0ac54

https://github.com/Expensify/App/assets/93399543/edf19e3f-b5b4-4aee-b6b1-ede9e601054f

https://github.com/Expensify/App/assets/93399543/973468e6-68ed-451d-b0ff-f7587a8796f4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1696669199917159

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by esh-g** [comment]
# Proposal

## Please re-state the problem we are trying to solve:
'task for' task copy words not translated on copy

## What is the root cause of this problem?
The root cause is that the task message is hard coded here:
https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/libs/actions/Task.js#L924

Moreover, currently how we handle the copying of the right text for task report action is completely different from the way we handle the copying of actions like an `IOU` or `reportPreview`. For task, we are directly manipulating the `messageHTML` here:
https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/pages/home/report/ContextMenu/ContextMenuActions.js#L190
But for things like `IOU` and `reportPreview` we are using the if-else ladder with their util functions.

This is actually related to https://github.com/Expensify/App/issues/29035 and a proposal for that issue will interfere with this one. According to the expected changes in that issue, mentioned in [this proposal](https://github.com/Expensify/App/issues/29035#issuecomment-1751461506), a `TaskUtils.js` will be created with a `getTaskTitle` function and the logic for getting the message got created task action will be decoupled from the `Closed/Re-opened/Cancelled` task action.

## What changes should be made to fix this?
We should first put this issue on hold for https://github.com/Expensify/App/issues/29035 and then if the expected changes are to be made in the that issue, the following changes can be made here:

**First**
We need to obviously add translations to `en.ts` and `es.ts` here:
https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/languages/en.ts#L1562-L1564
under `task.messages` we can create a new `created` field:
```js
created: ({taskTitle}: TaskCreatedActionParams) => `task for ${taskTitle}`,
```
same can be done for spanish ``tarea para ${taskTitle}``

**Second**
We then need to use those translations in `ContextMenuActions.js` where the hard coded `task for ${taskTitle}` will end up after fixing the fore-mentioned issue to be something like this:
```js
} else if (ReportActionsUtils.isCreatedTaskReportAction(reportAction)) {
   const taskTitle = TaskUtils.getTaskTitle(reportAction.childReportID, reportAction.childReportName);
   const translatedMessage = Localize.translateLocal('task.messages.created', {taskTitle});
   Clipboard.setString(translatedMessage);
}
```

**Third**
We should also add the translation for the task created message here:
https://github.com/Expensify/App/blob/e0ab03828a60d18b7a6c7a966a01fdfae3a6ecb4/src/libs/actions/Task.js#L73

**Comment by AmjedNazzal** [comment]
same RC as this [post](https://github.com/Expensify/App/issues/26621#event-10360036119) which was decided not to be fixed, if it will be fixed I think it's fair to re-open that post.

**Comment by dhanashree-sawant** [comment]
We are hardcoded text in this case, not sure if it was same in that issue too.

**Comment by AmjedNazzal** [comment]
All text in the app except for errors and user input is hardcoded, It's the same thing in both posts, at the moment we are not supporting back-end localization for these and in the post I linked it was decided to put all related issues on hold until a back-end localization is introduced because they didn't want a front-end workaround.

**Comment by burczu** [comment]
I don't agree with @AmjedNazzal that this is related to the lack of localization on the backend side. The text we use here is indeed hardcoded and it shouldn't be. But I agree with the statement from the @aswin-s's [proposal](https://github.com/Expensify/App/issues/29082#issuecomment-1752844050):

> This is actually related to https://github.com/Expensify/App/issues/29035 and a proposal for that issue will interfere with this one. According to the expected changes in that issue, mentioned in https://github.com/Expensify/App/issues/29035#issuecomment-1751461506, a TaskUtils.js will be created with a getTaskTitle function and the logic for getting the message got created task action will be decoupled from the Closed/Re-opened/Cancelled task action.

That's why I agree we should put this on hold until the mentioned issue is resolved and then check if the issue still occurs.

**Comment by isabelastisser** [comment]
Hey @burczu, should we close/put this issue on hold based on this [comment?](https://github.com/Expensify/App/issues/29082#issuecomment-1757363427)

**Comment by isabelastisser** [comment]
Bump @burczu , can you please follow up? Thanks!

**Comment by burczu** [comment]
@isabelastisser I think we should put this one on hold, because [this issue](https://github.com/Expensify/App/issues/29035) could fix it (it waits for some BE changes now).
